src/bucket/BucketManagerImpl.cpp

```diff
@@ -867,4 +868,165 @@ BucketManagerImpl::isShutdown() const
 {
     return mIsShutdown;
 }
+
+static void
+scanBucket(std::string const& pass, size_t sz, BucketInputIterator& in,
```

Comment:
If you are going to keep this function (see my other suggestion about reusing VerifyBucketWork functionality), I think its signature could be improved. Specifically, it could take phase name, a bool to indicate whether we need to compute hash, bucket pointer and an optional lambda. Then the caller doesn't need to worry about size, and creating the right `BucketInputIterator`

Reply:
Ok. Had to tweak this slightly (passing optional<Hash> and returning bool) but it seems to work ok.